File: sdk/cpp/src/api/dns.cpp
// Copyright 2026 Hypermesh Foundation. All rights reserved.
// Licensed under the Business Source License 1.1.

#include <hypermesh/api/dns.hpp>

namespace hypermesh {

DnsList DnsApi::list() const {
    auto j = http_.get("/api/v1/dns/list");
    return j.get<DnsList>();
}

DnsRecord DnsApi::resolve(const std::string& name) const {
    auto j = http_.get("/api/v1/dns/resolve/" + name);
    return j.get<DnsRecord>();
}

DnsRegisterResponse DnsApi::register_record(const std::string& name,
                                             const std::string& address) const {
    auto body = nlohmann::json{{"name", name}, {"address", address}};
    auto j = http_.post("/api/v1/dns/register", body);
    return j.get<DnsRegisterResponse>();
}

} // namespace hypermesh

File: sdk/typescript/src/api/dns.ts
import type { HttpClient } from "../client.js";
import type {
  DnsListResponse,
  DnsRegisterResponse,
  DnsResolveResponse,
} from "../types.js";

export class DnsApi {
  constructor(private readonly http: HttpClient) {}

  async list(): Promise<DnsListResponse> {
    return this.http.get<DnsListResponse>("/api/v1/dns/list");
  }

  async resolve(name: string): Promise<DnsResolveResponse> {
    return this.http.get<DnsResolveResponse>(
      `/api/v1/dns/resolve/${encodeURIComponent(name)}`,
    );
  }

  async register(name: string, address: string): Promise<DnsRegisterResponse> {
    return this.http.post<DnsRegisterResponse>("/api/v1/dns/register", {
      name,
      address,
    });
  }
}
